src/ICollection.ts

```diff
@@ -0,0 +1,80 @@
+import Promise = Q.Promise;
+import {DistributedObject} from './DistributedObject';
+/**
+ * ICollection uses serialized representation of objects to compare them.
+ */
+export interface ICollection<E> extends DistributedObject {
+    /**
+     * Ensures that this collection has given item
+     * @param item
+     * @throws {Error} if item is null or undefined.
+     * @return true if this collection did not contain the element before.
+     */
+    add(item : E) : Promise<boolean>;
+
+    /**
+     * Ensures that this collection all the items in given collection.
+     * This collection contains all elements of collection and its previous items at the end.
+     * @param collection
+     * @throws {Error} if collection or one of its elements is null or undefined.
+     * @return true if this collection changed, false otherwise.
+     */
+    addAll(collection : ICollection<E>) : Promise<boolean>;
```

Comment:
All ICollection parameters in this interface are ordinary, some language builtin collection, what ever exist in TypeScript but not ICollection.